### PR TITLE
Blog App: Validation & Specs

### PR DESCRIPTION
**Pull request by jasonfelice**
# Rails Blog App: Milestone 3

## Validation & Specs

- [x] Added the following validations:
  - For the User model:
    - Name must not be blank.
    - PostsCounter must be an integer greater than or equal to zero.
  - For the Post model:
    - Title must not be blank.
    - Title must not exceed 250 characters.
    - CommentsCounter must be an integer greater than or equal to zero.
    - LikesCounter must be an integer greater than or equal to zero.
 - [x] Added unit specs for all of your models' methods and validations.

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  before { subject.save }
+
+  it 'name should be present' do
+    subject.name = nil
+    expect(subject).to_not be_valid
+  end
+
+  it 'Should be valid' do
+    expect(subject).to be_valid
+  end
```
Comment (TedLivist) [inline spec/models/user_spec.rb:15]: - [x] Kindly include tests for the `posts_counter` numericality validation as contained in the user model.

- [x] All custom methods should be tested. Kindly include tests for the `recent_posts` method as contained in the user model.

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  end
+
+  it 'Should be valid' do
+    expect(subject).to be_valid
+  end
+
+  it 'title should be present' do
+    subject.title = nil
+    expect(subject).to_not be_valid
+  end
```
Comment (TedLivist) [inline spec/models/post_spec.rb:23]: - [x] Kindly include tests for the numericality validations of the `comments_counter` and `likes_counter` as contained in the Post model.

- [x] Include tests for the title `length` validation.

- [x] Kindly add tests for the `recent_comments` method as contained in the Post model

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @jasonfelice,

Good job so far!
✔️ Tests are passing
✔️ Good documentation
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by brytebee — APPROVED** [review]
Hi, @jasonfelice 👋,

##

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- Well-written tests and all tests pass.✔️
- App functions as it should.✔️

##

# STATUS: APPROVED 🕺🥳 

Congratulations🍾🥳!! Your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 



Good luck! 🎉 

##

###  **Optional Suggestions:**
- N/A

------

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**